test/puppetlabs/puppetdb/http/facts_test.clj

```diff
@@ -533,22 +533,26 @@
                              :values facts1
                              :timestamp (now)
                              :environment "DEV"
-                             :producer_timestamp (now)})
+                             :producer_timestamp (now)
+                             :producer "bar1"})
       (scf-store/add-facts! {:certname  "foo2"
                              :values facts2
                              :timestamp (now)
                              :environment "DEV"
-                             :producer_timestamp (now)})
+                             :producer_timestamp (now)
+                             :producer "bar2"})
       (scf-store/add-facts! {:certname "foo3"
                              :values facts3
                              :timestamp (now)
                              :environment "DEV"
-                             :producer_timestamp (now)})
+                             :producer_timestamp (now)
+                             :producer "bar3"})
       (scf-store/add-facts! {:certname "foo4"
                              :values facts4
                              :timestamp (now)
                              :environment "DEV"
-                             :producer_timestamp (now)})
+                             :producer_timestamp (now)
+                             :producer "bar4"})
```

Comment:
might be good to mix in a case of a nil producer somewhere -- I don't think we need to be too crazy about it but we've hit upgrade bugs around stuff like that before.